### PR TITLE
Modify Helix-related targets scripts to handle new conventions

### PR DESCRIPTION
**Pull request by mellinoe**
There's a few parts to this, and the rest of it is still a WIP. Putting this out to get some feedback from @karajas and @MattGal on the best way to move forward.

* I've added back a trimmed-down version of the Runner.* scripts and templates. We no longer need to copy and files from the package cache, but we do need to pass in a "Runtime Path" instead. This is essentially the `bin/runtime/netcoreapp-xyz` directory. This will be zipped up and delivered much in the same way that the packages directory is nowadays.
* Local test execution now uses those runner scripts like they did before
* This is still using CoreRun, but using dotnet shouldn't be much work on top of this, just changing which folders get zipped up and some paths changed.
* I haven't validated the zip files and made any changes to Helix yet.

### CODE REVIEW

**Comment by mellinoe** [comment]
Changes since the initial commit:

* Actually respect `TestDisabled` property in tests.targets. This got messed up in my refactoring/trimming earlier
* Hook in the `ArchiveTestBuild` target properly. This is the target that archives individual test folders
* Add a `CompressRuntimeDir` target, which is essentially the replacement for the `CompressPackagesDir` target. This just compresses the entire runtime folder into a "test-runtime-$(BuildConfiguration).zip" file.

I've validated that I'm able to unzip a test project's archive, and the test-runtime archive, and run tests successfully. I've not done any more "real" testing with Helix itself.

@karajas @MattGal

File: Tools-Override/CloudTest.targets [hunk 1]
```diff
@@ -0,0 +1,514 @@
...
+    <PerfTestListFile>$(TestWorkingDir)$(OSPlatformConfig)/$(PerfTestListFilename)</PerfTestListFile>
+    <BuildStatsJsonFile>$(TestWorkingDir)$(OSPlatformConfig)/BuildStats.json</BuildStatsJsonFile>
+    <RunnerScript Condition="'$(RunnerScript)' == ''">$(TestRunnerScript)</RunnerScript>
+    <SupplementalPayloadDir Condition="'$(SupplementalPayloadDir)' == ''">$(TestWorkingDir)SupplementalPayload/</SupplementalPayloadDir>
+    <SupplementalPayloadFilename>SupplementalPayload.zip</SupplementalPayloadFilename>
+    <SupplementalPayloadFile>$(ArchivesRoot)$(SupplementalPayloadFilename)</SupplementalPayloadFile>
+
+    <!-- Workaround to bring the whole Test ILC folder if required  -->
+    <TestILCVersion>1.4.24208-prerelease</TestILCVersion>
+    <TestILCFolder>$(PackagesDir)Microsoft.DotNet.TestILC\$(TestILCVersion)\contentFiles\any\any\TestILC</TestILCFolder>
```
Comment (MattGal) [inline Tools-Override/CloudTest.targets]: Does the previous model of restoring full UWP .NET Core dependencies work in the new world?  (i.e. do the other TFMs stay the same)

Comment (mellinoe) [inline Tools-Override/CloudTest.targets]: At this point, none of the other targets than .NET Core are functional at all in dev/eng. We know there's going to be some things that are broken that we will need to fix after .NET Core is working end-to-end.

File: Tools-Override/CloudTest.targets [hunk 2]
```diff
@@ -0,0 +1,514 @@
...
+      </UnixPlatformTestArchives>
+    </ItemGroup>
+
+    <PropertyGroup>
+        <OSPlatformTestArchivesListFile>$(TestArchivesRoot)/$(OSPlatformConfig)-Test-Archive-List.txt</OSPlatformTestArchivesListFile>
+        <AnyOSPlatformTestArchivesListFile>$(AnyOSTestArchivesRoot)/$(AnyOSPlatformConfig)-Test-Archive-List.txt</AnyOSPlatformTestArchivesListFile>
+        <UnixPlatformTestArchivesListFile>$(UnixTestArchivesRoot)/$(UnixPlatformConfig)-Test-Archive-List.txt</UnixPlatformTestArchivesListFile>
+    </PropertyGroup>
+
+    <!-- This is where we craft our archive Listes. These will sit at the base of the uploaded package. -->
```
Comment (MattGal) [inline Tools-Override/CloudTest.targets]: brought back a typo

File: Tools-Override/CloudTest.targets [hunk 3]
```diff
@@ -0,0 +1,514 @@
...
+    <HelixApiEndpoint Condition="'$(_UseUpdatedHelixApi)'=='true'">https://helix.dot.net/api/2016-06-28/jobs</HelixApiEndpoint>
+    <HelixApiEndpoint Condition="'$(HelixApiEndpoint)'==''">https://helix.dot.net/api/jobs</HelixApiEndpoint>
+  </PropertyGroup>
+
+  <Import Project="$(MSBuildThisFileDirectory)CloudTest.Perf.targets" Condition="'$(Performance)' == 'true'" />
+
+  <!-- main entrypoint -->
+  <Target Name="CloudBuild"
+          AfterTargets="Build"
+          DependsOnTargets="CompressRuntimeDir" />
```
Comment (MattGal) [inline Tools-Override/CloudTest.targets]: Why do you no longer need to depend on the other targets (VerifyInputs, PreCloudBuild, and UploadContent specifically)?

File: Tools-Override/RunnerTemplate.Unix.txt [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+  # Only copy the file over if the directory is empty. Otherwise, do nothing.
+  if [ ! "$(ls -A $storage_location)" ]; then 
+    echo "Copying core file $core_file_name to $storage_location"
+    cp $core_file_name $storage_location
+  fi
+}
+
+if [ "$RUNTIME_PATH" == "" ]
+then
+echo error: RUNTIME_PATH is not defined.  Usage: $0 RUNTIME_PATH ExecutionDir
```
Comment (MattGal) [inline Tools-Override/RunnerTemplate.Unix.txt]: We don't actually honor ExecutionDir here, we just run wherever the script is.

File: Tools-Override/RunnerTemplate.Windows.txt [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+:: ========================= END Test Execution =================================
+
+:ShowUsage
+echo.
+echo Usage:
+echo.
+echo %0 {runtime path}
+echo.
+echo Parameters:
+echo runtime rath :        (Mandatory) Root path containing the full test runtime, containing CoreCLR and all of the framework assemblies.
```
Comment (MattGal) [inline Tools-Override/RunnerTemplate.Windows.txt]: Nit: doesn't really need the 2nd "containing"

File: Tools-Override/tests.targets [hunk 6]
```diff
@@ -27,8 +28,11 @@
 
   <PropertyGroup>
     <TestRuntimeDir>$(RuntimePath)</TestRuntimeDir>
-    <TestHostExecutable Condition="'$(TestHostExecutable)' == ''">$(TestRuntimeDir)\corerun</TestHostExecutable>
-    <XunitExecutable Condition="'$(XunitExecutable)' == ''">$(TestRuntimeDir)/xunit.console.netcore.exe</XunitExecutable>
+    <TestRuntimeEnvVar Condition="'$(OS)' == 'Windows_NT'">%RUNTIME_PATH%\</TestRuntimeEnvVar>
+    <TestRuntimeEnvVar Condition="'$(OS)' != 'Windows_NT'">$RUNTIME_PATH/</TestRuntimeEnvVar>
+    <TestHostExecutable Condition="'$(TestHostExecutable)' == '' And '$(OS)' == 'Windows_NT'">$(TestRuntimeEnvVar)corerun.exe</TestHostExecutable>
```
Comment (MattGal) [inline Tools-Override/tests.targets]: Not in this PR but you've changed the meaning of this variable, might be worth a rename to something like TestHostExecutablePath

**Comment by karajas** [comment]
The only change I'd be worried about is the path to the tests drops given that TestTFM doesn't exist anymore.

File: Tools-Override/CloudTest.targets (hunk 3 above)
Comment (mellinoe) [inline Tools-Override/CloudTest.targets]: That was actually a temporary change I accidentally committed, my mistake.

File: Tools-Override/RunnerTemplate.Unix.txt (hunk 4 above)
Comment (mellinoe) [inline Tools-Override/RunnerTemplate.Unix.txt]: Fixing

**Comment by mellinoe** [comment]
@karajas Yeah, there is still some TestTFM stuff that should be cleaned out of the targets. We can do that separately, though; I'd like to remove all of it at once to make sure nothing inadvertently breaks from its removal. I think that all of the stuff using `$(TestTFM) in CloudTest.targets should be harmless if the property is never set.

**Pull request merged**